org.osgi.test.common/src/main/java/org/osgi/test/common/service/ServiceAware.java

```diff
@@ -21,9 +21,10 @@
 
 import org.osgi.annotation.versioning.ProviderType;
 import org.osgi.framework.ServiceReference;
+import org.osgi.test.common.tracking.TrackingConfig;
 
 @ProviderType
-public interface ServiceAware<T> {
+public interface ServiceAware<T> extends TrackingConfig {
```

Comment:
I know this extends helps you testing but it seems the wrong abstraction. For the testing can't we just inject a ServiceAware object and cast to BaseServiceUse or TrackingConfig since we know about our impl?

Reply:
There was no reason to hide this interface from the end user. In fact it contains details that are quite useful for things like the assertj's diagnostic `.as(...)' function.

Reply:
> There was no reason to hide this interface from the end user. In fact it contains details that are quite useful for things like the assertj's diagnostic `.as(...)' function.

OK, Then why not just put those methods in ServiceAware and get rid of TrackingConfig? I don't see any separate need for TrackingConfig then.

Reply:
because they are implemented at different layers.

Reply:
I don't see that. TrackServices implements it for no real reason. BaseServiceUse implements it (as well as ServiceAware) and delegates the methods to a TrackServices instance. So if you just move the methods from TrackingConfig to ServiceAware, you do not need TrackingConfig.

Reply:
I pushed a change with cast instead.